test/link/get_from_test.cpp

```diff
@@ -0,0 +1,39 @@
+#include <gtest/gtest.h>
+
+#include "utils.hpp"
+
+namespace test {
+
+TEST_F(LinkTest, SimpleFrom) {
+    std::shared_ptr<sim::IRoutingDevice> src =
+        std::make_shared<DeviceMock>(DeviceMock());
+    std::shared_ptr<sim::IRoutingDevice> dst =
+        std::make_shared<DeviceMock>(DeviceMock());
+
+    auto link = sim::Link(src, dst);
+    ASSERT_EQ(link.get_from(), src);
```

Comment:
Maybe userful to add 
```suggestion
    ASSERT_EQ(link.get_from(), src);
        ASSERT_EQ(link.get_to(), dsr);
```